src/SIL.XForge.Scripture/Services/MachineApiService.cs

```diff
@@ -1063,24 +1063,37 @@ CancellationToken cancellationToken
 
         try
         {
-            // Get the last completed build
-            TranslationBuild? translationBuild = (
-                await translationEnginesClient.GetAllBuildsAsync(translationEngineId, cancellationToken)
-            )
+            // Get the last build and the last completed build
+            IList<TranslationBuild> translationBuilds = await translationEnginesClient.GetAllBuildsAsync(
+                translationEngineId,
+                cancellationToken
+            );
+            TranslationBuild? lastTranslationBuild = translationBuilds.LastOrDefault();
+            TranslationBuild? lastCompletedTranslationBuild = translationBuilds
                 .Where(b => b.State == JobState.Completed)
                 .MaxBy(b => b.DateFinished);
-            if (translationBuild is not null)
+
+            // See if the very last build has completed
+            if (lastTranslationBuild?.State == JobState.Completed)
             {
                 // See if the current scripture range has changed
-                string currentScriptureRange = GetDraftedScriptureRange(translationBuild);
-                if (project.TranslateConfig.DraftConfig.CurrentScriptureRange != currentScriptureRange)
+                string currentScriptureRange = GetDraftedScriptureRange(lastTranslationBuild);
+                if (
+                    !string.IsNullOrWhiteSpace(currentScriptureRange)
+                    && project.TranslateConfig.DraftConfig.CurrentScriptureRange != currentScriptureRange
```

Comment:
Actually, how would this be possible? If `ExecuteWebhookAsync` happens, and we mark a build as completed and run `RetrievePreTranslationStatusAsync` (from `ExecuteWebhookAsync`), wouldn't we have updated `project.TranslateConfig.DraftConfig.CurrentScriptureRange` from that completed build? Are we doing this because maybe we might not have heard back from a webhook, and so here we tidy up in case we had never heard back?